pkg/placement/what-resolver_test.go

```diff
@@ -126,36 +122,39 @@ func TestWhatResolver(t *testing.T) {
-	setFakeClusterAPIResources(fakeKubeClusterClientset.Fake, wds1N.Path(), []*metav1.APIResourceList{
-		{GroupVersion: k8sCoreGroupVersion.String(),
-			APIResources: []metav1.APIResource{nsResource, cmResource},
-		}})
-	fcd := FakeClusterDisco{fakeKubeClusterClientset.Discovery(), fakeKubeClusterClientset}
+	// fakeKubeClusterClientset := fakekube.NewSimpleClientset(ns1, cm1)
+	// k8sCoreGroupVersion := metav1.GroupVersion{Version: "v1"}
+	// usualVerbs := []string{"get", "list", "watch"}
+	// nsResource := metav1.APIResource{Name: "namespaces", SingularName: "namespace", Namespaced: false, Version: "v1", Kind: "Namespace", Verbs: usualVerbs}
+	// cmResource := metav1.APIResource{Name: "configmaps", SingularName: "configmap", Namespaced: true, Version: "v1", Kind: "ConfigMap", Verbs: usualVerbs}
+	// setFakeClusterAPIResources(fakeKubeClusterClientset.Fake, wds1N.Path(), []*metav1.APIResourceList{
+	// 	{GroupVersion: k8sCoreGroupVersion.String(),
+	// 		APIResources: []metav1.APIResource{nsResource, cmResource},
+	// 	}})
+	//fcd := FakeClusterDisco{fakeKubeClusterClientset.Discovery(), fakeKubeClusterClientset}
 	scheme := machruntime.NewScheme()
 	orDie(apiextensionsv1.AddToScheme(scheme))
 	orDie(kcpapisv1alpha1.AddToScheme(scheme))
 	orDie(k8scorev1.AddToScheme(scheme))
-	fakeKCPClient := fakeclusterkcp.NewSimpleClientset()
-	bindingFactory := bindingfactory.NewSharedInformerFactory(fakeKCPClient, 0)
-	bindingClusterPreInformer := bindingFactory.Apis().V1alpha1().APIBindings()
+	//fakeKCPClient := fakeclusterkcp.NewSimpleClientset()
+	//bindingFactory := bindingfactory.NewSharedInformerFactory(fakeKCPClient, 0)
+	//bindingClusterPreInformer := bindingFactory.Apis().V1alpha1().APIBindings()
 	fakeDynamicClusterClientset := fakeclusterdynamic.NewSimpleDynamicClient(scheme, ns1, cm1, ns2, cm2, cm3)
 	dynamicClusterInformerFactory := clusterdynamicinformer.NewDynamicSharedInformerFactory(fakeDynamicClusterClientset, 0)
 	// crdClusterPreInformer := dynamicClusterInformerFactory.ForResource(apiextensionsv1.SchemeGroupVersion.WithResource("customresourcedefinitions"))
 	// bindingClusterPreInformer := dynamicClusterInformerFactory.ForResource(kcpapisv1alpha1.SchemeGroupVersion.WithResource("apibindings"))
 
-	fakeApiExtClusterClientset := fakeapiext.NewSimpleClientset()
-	apiExtClusterFactory := apiextinfact.NewSharedInformerFactory(fakeApiExtClusterClientset, 0)
-	crdClusterPreInformer := apiExtClusterFactory.Apiextensions().V1().CustomResourceDefinitions()
+	//fakeApiExtClusterClientset := fakeapiext.NewSimpleClientset()
+	//apiExtClusterFactory := apiextinfact.NewSharedInformerFactory(fakeApiExtClusterClientset, 0)
+	//crdClusterPreInformer := apiExtClusterFactory.Apiextensions().V1().CustomResourceDefinitions()
 
 	fakeUpKubeClient := fakeupkube.NewSimpleClientset()
 	kbSpaceRelation := kbuser.NewKubeBindSpaceRelation(ctx, fakeUpKubeClient)
+	spaceProviderNs := "spaceprovider-fake"
+	// TODO fake
```

Comment:
This test is broken.

```
mspreitz@mjs13 placement % go test .
I1207 23:15:46.647214   29717 set-binder-test.go:88] "Setting ResolvedWhat" epRef={"Cluster":"wm1","Name":"ep1"} resolvedWhat={"Downsync":[{"Key":{"First":{"group":"apiextensions.k8s.io","resource":"customresourcedefinitions"},"Second":"","Third":"crd1"},"Val":{"APIVersion":"v1","ReturnSingletonState":true,"CreateOnly":false}}],"Upsync":[{"apiGroup":"group1.test","resources":["sprockets","flanges"],"names":["George","Cosmo"]},{"apiGroup":"group2.test","resources":["cogs"],"names":["William"]}]}
I1207 23:15:46.647692   29717 set-binder-test.go:134] "Setting ResolvedWhat" epRef={"Cluster":"wm1","Name":"ep1"} resolvedWhat={"Downsync":[{"Key":{"First":{"group":"","resource":"namespaces"},"Second":"","Third":"ns-a"},"Val":{"APIVersion":"v1","ReturnSingletonState":false,"CreateOnly":true}}],"Upsync":[{"apiGroup":"group1.test","resources":["sprockets","flanges"],"names":["George","Cosmo"]},{"apiGroup":"group3.test","resources":["widgets"],"names":["*"]}]}
I1207 23:15:46.647749   29717 set-binder-test.go:137] "Adding resource discovery mapping" key="(wm1, { namespaces})" val={"Namespaced":true,"SupportsInformers":true,"PreferredVersion":"v1"}
--- FAIL: TestWhatResolver (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x101b6afd0]

goroutine 89 [running]:
testing.tRunner.func1.2({0x1024f4820, 0x103a4f570})
	/opt/homebrew/Cellar/go@1.19/1.19.13/libexec/src/testing/testing.go:1396 +0x1c8
testing.tRunner.func1()
	/opt/homebrew/Cellar/go@1.19/1.19.13/libexec/src/testing/testing.go:1399 +0x378
panic({0x1024f4820, 0x103a4f570})
	/opt/homebrew/Cellar/go@1.19/1.19.13/libexec/src/runtime/panic.go:884 +0x204
github.com/kubestellar/kubestellar/space-framework/pkg/client/clientset/versioned.NewForConfig(0x100c4d060?)
	/Users/mspreitz/go/src/github.com/kubestellar/kubestellar/space-framework/pkg/client/clientset/versioned/clientset.go:63 +0x20
github.com/kubestellar/kubestellar/space-framework/pkg/msclientlib.NewMultiSpace({0x102934488?, 0x140003278c0}, 0x102955ec0?)
	/Users/mspreitz/go/src/github.com/kubestellar/kubestellar/space-framework/pkg/msclientlib/client.go:129 +0xfc
github.com/kubestellar/kubestellar/pkg/placement.TestWhatResolver(0x14000602820)
	/Users/mspreitz/go/src/github.com/kubestellar/kubestellar/pkg/placement/what-resolver_test.go:155 +0x1014
testing.tRunner(0x14000602820, 0x1028fe820)
	/opt/homebrew/Cellar/go@1.19/1.19.13/libexec/src/testing/testing.go:1446 +0x10c
created by testing.(*T).Run
	/opt/homebrew/Cellar/go@1.19/1.19.13/libexec/src/testing/testing.go:1493 +0x300
FAIL	github.com/kubestellar/kubestellar/pkg/placement	0.734s
FAIL
```

Reply:
There is no framework or exemplar for fake `*rest.Config`. If #1271 were merged and used here, then fake clientsets could be used instead.